docs/en/guide/configurations.md

```diff
@@ -142,54 +145,237 @@ globalConfig:
+    - cache-size=204800
+  ...
+```
+
+#### Via StorageClass definition (deprecated) {#dynamic-mount-options}
+
+Customize mount options in `StorageClass` definition. If you need to use different mount options for different applications, you'll need to create multiple `StorageClass`, each with different mount options.
+
+Due to StorageClass being the template used for creating PVs, **modifying mount options in StorageClass will not affect existing PVs**, if you need to adjust mount options for dynamic provisioning, you'll have to delete existing PVCs, or [directly modify mount options in existing PVs](#static-mount-options).
+
+```yaml {6-7}
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: juicefs-sc
+provisioner: csi.juicefs.com
+mountOptions:
+  - cache-size=204800
+parameters:
+  ...
+```
+
+### Health check & pod lifecycle {#custom-probe-lifecycle}
+
+Minimum required version is CSI Driver v0.24.0. Upon modification, application pods or mount pods need to be re-created for changes to take effect. If you decide to re-create mount pod, be sure to enable [automatic mount point recovery](./configurations.md#automatic-mount-point-recovery) in advance, to avoid permanent loss of mount point within the application pod.
+
+Targeted scenarios:
+
+- Use `readinessProbe` to set up mount pod healthchecking, for monitoring and alerting;
+- Customize `preStop` in sidecars to guarantee mount container exit order (mount container exit after application container), refer to [sidecar recommendations](../administration/going-production.md#sidecar) for details.
 
 ```yaml
-juicefs/mount-cpu-limit: ""
-juicefs/mount-memory-limit: ""
-juicefs/mount-cpu-request: ""
-juicefs/mount-memory-request: ""
+  - pvcSelector:
+      matchLabels:
+        custom-probe: "true"
+    readinessProbe:
+      exec:
+        command:
+        - stat
+        - ${MOUNT_POINT}/${SUB_PATH}
+      failureThreshold: 3
+      initialDelaySeconds: 10
+      periodSeconds: 5
+      successThreshold: 1
+```
 
-juicefs/mount-labels: ""
-juicefs/mount-annotations: ""
-juicefs/mount-service-account: ""
-juicefs/mount-image: ""
-juicefs/mount-delete-delay: ""
+### Mount extra volumes {#custom-volumes}
 
-# Clean cache at mount pod exit
-juicefs/clean-cache: ""
-juicefs/mount-cache-pvc: ""
-juicefs/mount-cache-emptydir: ""
-juicefs/mount-cache-inline-volume: ""
+Targeted scenarios:
 
-# Mount the hosts file or directory to pod
-# Container mount path will be the same as host path, this doesn't support customization
-juicefs/host-path: "/data/file.txt"
+- Some object storage providers (like Google Cloud Storage) requires extra credential files for authentication, this means you'll have to create a separate Secret to store these files, and reference it in volume credentials (JuiceFS-secret in below examples), so that CSI Driver will mount these files into the mount pod. The relevant environment variable needs to be added to specify the added files for authentication.
```

Comment:
```suggestion
- Some object storage providers (like Google Cloud Storage) require extra credential files for authentication. This means you will have to create a separate Secret to store these files and reference it in volume credentials (JuiceFS-secret in below examples), so that CSI Driver will mount these files into the mount pod. The relevant environment variable needs to be added to specify the added files for authentication.
```